ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -334,15 +342,61 @@ public void onShutdown() {
                 .anyMatch(s -> s.equals("ssl")) ? "https" : "http";
 
             final int localPort = connector.getLocalPort();
-
-            return new URL(protocol, "localhost", localPort, "");
+            final String host = getListenerHostName(connector.getHost(), nets);
+            return new URL(protocol, host, localPort, "");
           } catch (final Exception e) {
             throw new RuntimeException("Malformed listener", e);
           }
         })
         .collect(Collectors.toList());
   }
 
+  private Enumeration<NetworkInterface> getNetworkInterfaces() {
+    Enumeration<NetworkInterface> nets;
+    try {
+      nets = NetworkInterface.getNetworkInterfaces();
+    } catch (SocketException e) {
+      throw new RuntimeException("Can't get network interfaces", e);
+    }
+    return nets;
+  }
+
+  /**
+   * Get a host address that will be broadcast to other instances and used to query the rest api.
+   * @param connectorHost The host as gotten from the jetty connector
+   * @return The host to broadcast to clients
+   */
+  @VisibleForTesting
+  String getListenerHostName(String connectorHost, Enumeration<NetworkInterface> nets)
```

Comment:
Rather than exposing this a `@VisitbleForTesting`, which not move it to a util class and add the tests specifically to the until class.

This would keep this class, which is already large, from getting even larger.